core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -251,7 +250,19 @@ public ServerBuilder port(InetSocketAddress localAddress, SessionProtocol protoc
      * @see <a href="#no_port_specified">What happens if no HTTP(S) port is specified?</a>
      */
     public ServerBuilder port(ServerPort port) {
-        ports.add(requireNonNull(port, "port"));
+        requireNonNull(port, "port");
+        for (int i = 0; i < ports.size(); i++) {
+            final ServerPort p = ports.get(i);
+            if (p.equals(port)) {
+                final Set<SessionProtocol> protocols =
+                        new ImmutableSet.Builder<SessionProtocol>().addAll(p.protocols())
+                                                                   .addAll(port.protocols())
+                                                                   .build();
```

Comment:
This will raise an exception when a user attempts this:

```java
sb.http(8080);
sb.http(8080);
```

Could we quietly deduplicate?

Reply:
It doesn't raise an exception when adding duplicate elements as Javadoc describes. :-)
```
    /**
     * Adds each element of {@code elements} to the {@code ImmutableSet}, ignoring duplicate
     * elements (only the first duplicate element is added).
     *
     * @param elements the {@code Iterable} to add to the {@code ImmutableSet}
     * @return this {@code Builder} object
     * @throws NullPointerException if {@code elements} is null or contains a null element
     */
```

Reply:
Oops! Thanks for education. It was `ImmutableMap` builder which disallowed a duplicate.